auth-app/internal/handler/keys.go

```diff
@@ -0,0 +1,82 @@
+package handler
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/labstack/echo"
+	"github.com/lalizita/streaming-key-server-manager/internal/model"
+	"github.com/lalizita/streaming-key-server-manager/internal/service"
+)
+
+type KeysHandler interface {
+	AuthStreamingKey(ctx echo.Context) error
+}
+
+type keysHandler struct {
+	KeysService service.KeysService
+}
+
+func NewHandler(s service.KeysService) *keysHandler {
+	return &keysHandler{
+		KeysService: s,
+	}
+}
+
+func (h *keysHandler) AuthStreamingKey(ctx echo.Context) error {
+	log.Default().Println("Auth...")
+	body := ctx.Request().Body
+	defer body.Close()
+	fields, _ := io.ReadAll(body)
+	log.Println("=====", string(fields))
+	authValues := getKeyValues(fields)
+
+	keys, err := h.KeysService.AuthStreamingKey(authValues.Name, authValues.Key)
+	if err != nil {
+		return ctx.JSON(http.StatusInternalServerError, "Error findind key")
+	}
+
+	if keys.Key != "" {
+		log.Default().Println("User authenticated")
+
+		log.Println("Name:", authValues.Name)
+		log.Println("Addr:", authValues.Addr)
+
+		// According to the nginx rtmp module the redirect url must use an IP address
+		// If not set the query parameter "name", at least in my tests, the name of the
+		// directory created for hls is not changed
+		newStreamURL := fmt.Sprintf("rtmp://%s:1935/hls-live/%s?name=%s", authValues.Addr, authValues.Name, authValues.Name)
+		log.Println("Redirecting to:", newStreamURL)
+
+		// Respond with a 302 redirect to the new stream URL
+		return ctx.Redirect(http.StatusFound, newStreamURL)
+	}
+
+	log.Default().Println("Forbidden User")
+	return ctx.String(http.StatusForbidden, "")
+}
+
+func getKeyValues(s []byte) model.Keys {
```

Comment:
Can't we just use the `url.Parse()` and/or `url.ParseQuery()` from `net/url` here? It would simplify this function a lot.

https://gobyexample.com/url-parsing

Reply:
I've made some changes to simplify this function in this [commit](https://github.com/lalizita/streaming-key-server-manager/pull/7/commits/82f489669c445ca5959c8b35c56cd1df6c4486ff). The use of `url.Parse()` would be to check the `newStreamUrl` or to get the param `name`?